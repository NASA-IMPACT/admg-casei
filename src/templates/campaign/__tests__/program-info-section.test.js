import React from "react"
import { create } from "react-test-renderer"

import ProgramInfoSection from "../program-info-section"

const testString = "test string"
const testUrl = "https://www.test-campaign.io"
const testImage = {
  shortname: "ACES",
  description: "logo",
  category: "logo",
  gatsbyImg: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "fullWidth",
        backgroundColor: "#080808",
        images: {
          fallback: {
            src:
              "/static/3d17ba6135efff1e39e6b4d97157a123/893c2/3b06165b-7033-4357-8f15-d2b68cbd84ff.png",
            srcSet:
              "/static/3d17ba6135efff1e39e6b4d97157a123/893c2/3b06165b-7033-4357-8f15-d2b68cbd84ff.png 129w",
            sizes: "100vw",
          },
          sources: [
            {
              srcSet:
                "/static/3d17ba6135efff1e39e6b4d97157a123/23c0e/3b06165b-7033-4357-8f15-d2b68cbd84ff.webp 129w",
              type: "image/webp",
              sizes: "100vw",
            },
          ],
        },
        width: 1,
        height: 0.9922480620155039,
      },
    },
  },
}

describe("Program Info Section", () => {
  it("renders logo when present in props", () => {
    const component = create(
      <ProgramInfoSection
        id="program-info"
        logoFullWidth={testImage}
        shortname="AirMOSS"
        fundingAgency={testString}
        fundingProgram={testString}
        programLead={testString}
        leadInvestigator={testString}
        dataManager={testString}
        repositoryWebsite={testUrl}
        partnerOrgListing={testString}
        publicationLink={testUrl}
      />
    )
    const tree = component.toJSON()
    const instance = component.root

    expect(instance.findAllByType("img")[0].props.alt).toBe(
      testImage.description
    )
    expect(tree).toMatchSnapshot()
  })
  it("renders placeholder when no logo is available", () => {
    const component = create(
      <ProgramInfoSection
        id="program-info"
        logo={{
          shortname: "GOES-R PLT",
          description: "GOES-R PLT logo",
          category: "logo",
          gatsbyImg: null,
        }}
        shortname={testString}
        fundingAgency={testString}
        fundingProgram={testString}
        programLead={testString}
        leadInvestigator={testString}
        dataManager={testString}
        repositoryWebsite={testUrl}
        partnerOrgListing={testString}
        publicationLink={testUrl}
      />
    )
    const tree = component.toJSON()
    const instance = component.root
    expect(instance.findByType("svg")).toBeDefined()
    expect(tree).toMatchSnapshot()
  })
})
