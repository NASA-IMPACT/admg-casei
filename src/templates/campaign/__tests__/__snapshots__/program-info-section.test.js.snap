// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Program Info Section renders logo when present in props 1`] = `
<section
  className="section__Container-sc-1oozlr5-0 rgaqP"
  data-cy="program-info-section"
  mode="darkTheme"
>
  <div
    className="section__Buffer-sc-1oozlr5-1 gccBEq"
    id="program-info"
  />
  <div
    className="sc-aXZVg fflPGZ"
    data-cy="program-info-section-header"
  >
    <a
      href="#program-info"
    >
      <h2>
        Program Info
      </h2>
    </a>
  </div>
  <div
    className="section__SectionContent-sc-1oozlr5-2 kkYoHL"
    mode="darkTheme"
  >
    <div
      className="sc-fqkvVR gOztfi"
    >
      <div
        className="sc-dcJsrY eCZDRB"
      >
        <div
          className="program-info-section___StyledDiv3-sc-celi18-0 URNyM"
          data-cy="campaign-logo"
        >
          <div
            className="gatsby-image-wrapper"
            data-gatsby-image-wrapper=""
            style={
              {
                "backgroundColor": undefined,
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              aria-hidden={true}
              style={
                {
                  "paddingTop": "99.2248062015504%",
                }
              }
            />
            <div
              aria-hidden={true}
              data-placeholder-image=""
              style={
                {
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            />
            <picture>
              <source
                data-srcset="/static/3d17ba6135efff1e39e6b4d97157a123/23c0e/3b06165b-7033-4357-8f15-d2b68cbd84ff.webp 129w"
                sizes="100vw"
                type="image/webp"
              />
              <img
                alt="logo"
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="/static/3d17ba6135efff1e39e6b4d97157a123/893c2/3b06165b-7033-4357-8f15-d2b68cbd84ff.png"
                data-srcset="/static/3d17ba6135efff1e39e6b4d97157a123/893c2/3b06165b-7033-4357-8f15-d2b68cbd84ff.png 129w"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
            <noscript>
              <picture>
                <source
                  sizes="100vw"
                  srcSet="/static/3d17ba6135efff1e39e6b4d97157a123/23c0e/3b06165b-7033-4357-8f15-d2b68cbd84ff.webp 129w"
                  type="image/webp"
                />
                <img
                  alt="logo"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/static/3d17ba6135efff1e39e6b4d97157a123/893c2/3b06165b-7033-4357-8f15-d2b68cbd84ff.png"
                  srcSet="/static/3d17ba6135efff1e39e6b4d97157a123/893c2/3b06165b-7033-4357-8f15-d2b68cbd84ff.png 129w"
                  style={
                    {
                      "backgroundColor": undefined,
                      "height": "100%",
                      "left": 0,
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transform": "translateZ(0)",
                      "transition": "opacity 250ms linear",
                      "width": "100%",
                      "willChange": "opacity",
                    }
                  }
                />
              </picture>
            </noscript>
            <script
              dangerouslySetInnerHTML={
                {
                  "__html": "const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}",
                }
              }
              type="module"
            />
          </div>
        </div>
      </div>
      <div
        className="content-group___StyledDiv-sc-gouaoh-0 Spbag"
      >
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Funding Agency
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Funding Program
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Funding Program Lead
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Principal Investigator(s)
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Data Manager / Technical Contact
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            NASA Data Repository
          </label>
          <div
            data-cy="program-info-content-text"
          >
            Currently unavailable
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Partner Organizations
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Overview Publication
          </label>
          <div
            data-cy="program-info-content-text"
          >
            Currently unavailable
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Program Info Section renders placeholder when no logo is available 1`] = `
<section
  className="section__Container-sc-1oozlr5-0 rgaqP"
  data-cy="program-info-section"
  mode="darkTheme"
>
  <div
    className="section__Buffer-sc-1oozlr5-1 gccBEq"
    id="program-info"
  />
  <div
    className="sc-aXZVg fflPGZ"
    data-cy="program-info-section-header"
  >
    <a
      href="#program-info"
    >
      <h2>
        Program Info
      </h2>
    </a>
  </div>
  <div
    className="section__SectionContent-sc-1oozlr5-2 kkYoHL"
    mode="darkTheme"
  >
    <div
      className="sc-fqkvVR gOztfi"
    >
      <div
        className="sc-dcJsrY eCZDRB"
      >
        <svg
          height="120"
          viewBox="0 0 65 69"
          width="120"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="#FFF"
            fillRule="nonzero"
          >
            <path
              d="M57.57 17.157c.759-1.875 1.239-3.589 1.239-4.93C58.809 5.484 53.299 0 46.526 0c-4.326 0-8.13 2.244-10.318 5.618l-.028.045a18.788 18.788 0 00-3.337-.346l-.028-.002v-.007h-.025C15.147 5.308.791 19.589.791 37.154.791 54.718 15.147 69 32.791 69H32.817l.03-.002c17.62-.03 31.944-14.267 31.944-31.765 0-7.608-2.711-14.6-7.221-20.076zM26.146 7.623c-2.938 2.043-5.529 5.445-7.517 9.677-2.37-.586-4.585-1.32-6.593-2.198 3.894-3.612 8.736-6.284 14.11-7.48zm-15.273 8.619c2.16.987 4.556 1.816 7.133 2.467a42.558 42.558 0 00-2.092 6.423 9.635 9.635 0 00-1.38-.11c-5.329 0-9.664 4.315-9.664 9.62 0 .521.095 1.116.259 1.758-1.426.01-2.28.023-2.793.035.22-7.83 3.43-14.925 8.537-20.193zm6.168 10.694c3.265 1.055 5.634 4.109 5.634 7.707 0 .488-.113 1.075-.299 1.722-.136.474-.315.981-.53 1.518-1.102 2.726-3.122 6.125-4.96 8.94a102.86 102.86 0 01-2.352 3.444C12.455 47.363 8.85 41.9 7.228 37.895A15.495 15.495 0 016.7 36.39c-.191-.656-.307-1.253-.307-1.747 0-4.47 3.652-8.105 8.141-8.105.355 0 .703.03 1.045.075.503.064.992.171 1.462.323zM2.331 37.933c.6-.013 1.725-.023 3.274-.03 2.1 5.67 8.013 13.698 8.317 14.108l.612.828.613-.828c.073-.098.467-.634 1.05-1.463a41.96 41.96 0 001.808 5.32c-2.577.652-4.973 1.48-7.133 2.467-5.152-5.315-8.374-12.49-8.54-20.402zm9.704 21.541c2.009-.876 4.224-1.612 6.594-2.197 1.988 4.232 4.579 7.5 7.517 9.543-5.375-1.195-10.216-3.733-14.111-7.346zm20.019 8.116c-4.745-.394-9.032-4.447-11.978-10.648 3.721-.803 7.777-1.248 11.978-1.296V67.59zm0-13.46c-4.424.05-8.7.529-12.614 1.398a41.213 41.213 0 01-2.119-6.612c1.998-2.955 4.831-7.461 6.149-11.033l8.584.009v16.237zm0-17.757c-2.986-.004-5.677-.006-8.106-.008.159-.63.25-1.211.25-1.723 0-4.323-2.88-7.987-6.829-9.196a40.949 40.949 0 012.07-6.398c3.915.871 8.19 1.35 12.615 1.4v15.925zm0-17.441c-4.202-.05-8.257-.494-11.978-1.298 2.946-6.2 7.233-10.254 11.978-10.648v11.946zM36.736 7.81a10.643 10.643 0 011.17-1.964c1.962-2.621 5.091-4.33 8.62-4.33 5.932 0 10.76 4.805 10.76 10.71 0 .998-.312 2.25-.829 3.645a28.177 28.177 0 01-.642 1.575c-1.417 3.215-3.628 6.911-5.605 9.964a145.462 145.462 0 01-3.684 5.413c-1.474-2.059-5.285-7.529-7.95-12.616-.258-.491-.504-.98-.737-1.46-1.227-2.534-2.072-4.874-2.072-6.52 0-1.576.353-3.068.969-4.417zm-3.159-.824v-.02c.71.048 1.188.147 1.736.294l-.015.031a12.095 12.095 0 00-1.054 4.936c0 1.737.8 4.093 1.978 6.624-.874.045-1.756.072-2.645.082V6.986zm0 13.462a66.297 66.297 0 003.362-.12c3.343 6.602 8.65 13.804 8.974 14.243l.613.828.613-.828c.125-.167.974-1.321 2.16-3.044.172 1.585.273 3.215.3 4.878-6.075-.014-11.39-.023-16.022-.03V20.448zm0 17.445c5.329.008 10.817.018 16.025.028-.076 6.595-1.338 12.679-3.409 17.6-3.915-.867-8.191-1.344-12.616-1.392V37.893zm0 17.752c4.183.047 8.222.486 11.93 1.282-2.943 6.13-7.21 10.134-11.93 10.525V55.645zm5.932 11.092c2.906-2.024 5.47-5.314 7.446-9.48l.05.011c2.324.573 4.499 1.29 6.478 2.143a30.487 30.487 0 01-13.974 7.326zm15.226-8.541s-.055.088-.082.075c-2.132-.963-4.491-1.772-7.025-2.41 2.126-5.072 3.419-11.254 3.497-17.936 4.366.01 8.496.02 12.123.03-.185 7.789-3.345 14.857-8.4 20.125l-.113.116zm8.327-21.759s-8.193-.022-11.94-.03c-.038-2.33-.22-4.6-.538-6.781 2.06-3.1 4.59-7.188 6.298-10.887l.002-.003c3.808 4.876 5.964 10.932 6.174 17.516a7.5 7.5 0 01.004.132c.02.06 0 .053 0 .053z"
            />
            <path
              d="M46.606 6a4.83 4.83 0 00-2.995 1.046 5.024 5.024 0 00-1.685 2.423 5.1 5.1 0 00-.246 1.56c0 .108.009.212.016.318a5.03 5.03 0 002.952 4.294 4.824 4.824 0 001.847.412c.038 0 .075.006.113.006 2.054 0 3.816-1.29 4.555-3.119a5.084 5.084 0 00.359-1.645c.004-.088.012-.175.012-.265C51.533 8.256 49.323 6 46.606 6zm3.26 5.993c-.411 1.449-1.71 2.517-3.26 2.517-.37 0-.72-.076-1.053-.187a3.467 3.467 0 01-2.358-3.294c0-.042.01-.082.012-.123a3.488 3.488 0 01.768-2.065 3.4 3.4 0 011.46-1.069 3.307 3.307 0 011.172-.225c1.592 0 2.921 1.124 3.296 2.631.068.274.115.556.115.85-.001.337-.063.657-.151.965zM16.698 30.61a3.88 3.88 0 00-1.454-.482c-.157-.02-.311-.049-.473-.049-2.195 0-3.98 1.822-3.98 4.062 0 .988.362 1.883.939 2.587.73.893 1.815 1.474 3.04 1.474 1.23 0 2.317-.584 3.048-1.481a4.074 4.074 0 00.932-2.58 4.062 4.062 0 00-2.052-3.531zm-.428 5.511c-.417.33-.933.534-1.5.534l-.022-.002c-1.348-.014-2.44-1.134-2.44-2.512 0-1.387 1.104-2.515 2.463-2.515.098 0 .19.018.286.03.56.067 1.066.315 1.447.7.45.456.73 1.088.73 1.785a2.52 2.52 0 01-.964 1.98z"
            />
          </g>
        </svg>
      </div>
      <div
        className="content-group___StyledDiv-sc-gouaoh-0 Spbag"
      >
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Funding Agency
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Funding Program
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Funding Program Lead
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Principal Investigator(s)
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Data Manager / Technical Contact
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            NASA Data Repository
          </label>
          <div
            data-cy="program-info-content-text"
          >
            Currently unavailable
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Partner Organizations
          </label>
          <div
            data-cy="program-info-content-text"
          >
            test string
          </div>
        </div>
        <div
          data-cy="program-info-content"
        >
          <label
            className="content-item___StyledLabel-sc-goubn5-0 dgQhVR"
            data-cy="program-info-content-label"
          >
            Overview Publication
          </label>
          <div
            data-cy="program-info-content-text"
          >
            Currently unavailable
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
